Major refactor around  form management, fee & payload calculations.

hbulgarini opened the pull request:
This PR is a major refactor around the following items needed to be solved:

1) Manage consistently input fields for `Transfer`, `Remark` &  `CustomCall` components. #205 #207

- Inputs moved to transaction reducer so evaluations can happen on reducer making it smarter.
- The evaluations are being handled from the `transactionContext` following the data-model approach which handles logic.

2) Much optimized payload and estimated fee calculations: #195 
 
-  Payload and Estimated fee are now calculated in just one call avoiding a lot of unnecessary re-renders and complexity coordinating all process.
- Hook `useEstimatedFeePayload` now contains the logic of `useTransactionPreparation`, `useTransactionType`, `usePayload` and `useEstimatedFee`.
- Debounce happening on `useEvaluateCalculation` hook that checks the inputs in one place reducing the boilerplate on each component.

3)  Moved logic from hooks to async javascript function hooks
- useDispatchGenericCall.ts -> genericCall (util/apiUtlis.ts)
- useTransactionType.ts -> getTransactionCallWeight (util/transactionUtils.ts)

I apologize in advance for the size of this PR. It was really hard to make it in chunks because every chunk depends on the other and all the parts need to be tested together.

tomusdrw reviewed (approved):
I'm assuming it works as you expect and it's the best possible version of the code you want to get in, so let's merge as-is and give it a round of tests on deployment and then address any potential issues.

hbulgarini commented on src/hooks/transactions/useEvaluateCalculation.ts:
```diff
@@ -0,0 +1,101 @@
+  const debouncedCustomCall = useDebounceState(customCallInput);
+  const debouncedWeightInput = useDebounceState(weightInput);
+
+  const prevReceiverAddress = usePrevious(receiverAddress);
+  const prevTransferAmount = usePrevious(debouncedTransferAmount);
+  const prevRemarkInput = usePrevious(debouncedRemarkInput);
+  const prevCustomCallInput = usePrevious(debouncedCustomCall);
+  const prevWeightInput = usePrevious(debouncedWeightInput);
+  const prevSenderAccount = usePrevious(senderAccount);
```
i agree ... i'm not 100% convinced ...This is working as expected in terms of the number of fee calculations so we could accept it as a first version ( i will see in the next days )
 i will take sometime to think another way less manual

hbulgarini commented:
@tomusdrw on commit [0e9d630](https://github.com/paritytech/parity-bridges-ui/pull/245/commits/0e9d630d66effdd51249e5f48fca00524678a0b6) i got super picky with all the async process regarding the fee estimation. For example, i have implemented a mechanism that if during the fee calculation a new calculation is triggered , it will batch the last calculation to run it as soon as the current one is finished. In case the user span multiples updates, the last one will be executed.
Also i have removed `useEvaluateCalculation` hook and now all the logic happens on the reducer using helpers.

Lastly, all the tests are working fine and i have performed a manual testing regarding the button enable and estimation fee message displayed on the UI.

Only known bug that i will address separately  is the one regarding #247 regarding remark inputs.
